dragonfly/actions/mouse/_talon.py

```diff
@@ -0,0 +1,123 @@
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#   Lesser General Public License for more details.
+#
+#   You should have received a copy of the GNU Lesser General Public
+#   License along with Dragonfly.  If not, see
+#   <http://www.gnu.org/licenses/>.
+#
+
+from talon import ctrl
+
+from ._base import BaseButtonEvent, MoveEvent
+
+
+#---------------------------------------------------------------------------
+# Functions and event delegate for getting and setting the cursor position.
+
+def get_cursor_position():
+    ctrl.mouse_pos()
+
+
+def set_cursor_position(x, y):
+    ctrl.mouse_move(x, y)
+    return True
+
+
+class MoveEventDelegate(object):
+    @classmethod
+    def get_position(cls):
+        return get_cursor_position()
+
+    @classmethod
+    def set_position(cls, x, y):
+        return set_cursor_position(x, y)
+
+
+# Set MoveEvent's delegate. This allows us to set platform-specific
+# functions for getting and setting the cursor position without having
+# to do a lot of sub-classing for no good reason.
+MoveEvent.delegate = MoveEventDelegate()
+
+#---------------------------------------------------------------------------
+# Pynput mouse button and wheel up/down flags.
+
+
+_buttons = {
+    'left': 0,
+    'right': 1,
+    'middle': 2,
+    'four': 3,
+    'five': 4,
+}
+def get_button(name):
+    return buttons.get(name, None)
+
+
+PLATFORM_BUTTON_FLAGS = {
+    # ((button, event_type), down)
+    # The inner pair is used here and below to be compatible with the
+    # original Windows flags.
+    "left":   (((0, 0), 1),  # down
+               ((0, 0), 0)),  # up
+    "right":  (((1, 0), 1),
+               ((2, 0), 0)),
+    "middle": (((2, 0), 1),
+               ((2, 0), 0)),
+
+    # We call these "four" and "five" because Windows calls them that.
+    # These buttons typically behave as browser back and forward media keys.
+    "four": (((3, 0), 1),
+             ((3, 0), 0)),
+    "five": (((4, 0), 1),
+             ((4, 0), 0)),
+}
+
+PLATFORM_WHEEL_FLAGS = {
+    # ((button, event_type), scroll_count)
+    "wheelup": (('up', 1), 3),
+    "stepup": (('up', 1), 1),
+    "wheeldown": (('down', 1), 3),
+    "stepdown": (('down', 1), 1),
+    "wheelright": (('right', 1), 3),
+    "stepright": (('right', 1), 1),
+    "wheelleft": (('left', 1), 3),
+    "stepleft": (('left', 1), 1),
+}
+
+
+#---------------------------------------------------------------------------
+# event classes.
+
+
+class ButtonEvent(BaseButtonEvent):
+
+    def execute(self, window):
+        for ((button, event_type), flag) in self._flags:
+            # Check if the button is unknown.
+            if button is None:
+                event_type_s = "button" if event_type == 0 else "scroll"
+                raise ValueError("Unsupported %s event" % event_type_s)
+
+            if event_type == 0:  # Button press event
+                ctrl.mouse_click(button, down=flag)
```

Comment:
There appears to be a bug somewhere here such that `Mouse("left").execute()` clicks down but doesn't release the click (the expected behavior without an "up" or "down" qualifier).

Reply:
That's an unrelated bug in Talon on Windows. The fix is already in beta and will be released as part of v0.2. As a workaround, you should be able to do:

```
if flag: ctrl.mouse_click(button, down=True)
else: ctrl.mouse_click(button, up=True)
```

Reply:
Not urgent to get this working on my machine, but I'll note that the latest published beta (talon-windows-108-0.1.5-51-g3121.exe) still has this bug.

Reply:
fixed by 3eee2e8